sway-core/src/type_system/info.rs

```diff
@@ -389,6 +393,10 @@ impl PartialEqWithEngines for TypeInfo {
                         .get(*l_trait_type_id)
                         .eq(&type_engine.get(*r_trait_type_id), engines)
             }
+            (Self::Ref(l_ty), Self::Ref(r_ty)) => type_engine
```

Comment:
Silly optimization but if type ids match you are done, right?

```
(l_ty.type_id == r_ty.type_id) || (
type_engine.get(l_ty.type_id).eq(&type_engine.get(r_ty.type_id), engines))
```

Reply:
Good catch! Actually, considering how often this is called, I don't see this optimization as silly at all! Although simple one, it was not used for any variant, so I've added it everywhere where it's applicable. Curious to see those perf graphs of ours dropping a pixel or two more :-) Jokes aside, good catch!